### PR TITLE
Adds TGS DMAPI

### PR DESCRIPTION
**Pull request by Cyberboss**
Spreading the reach of my trojan

Gives access to logging, git revision stuff, chat messages and chat [commands](https://github.com/tgstation/tgstation/blob/master/code/modules/admin/chat_commands.dm)

See tgs.dm for full caps

### CODE REVIEW

File: code/__DEFINES/tgs.config.dm [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#define TGS_EXTERNAL_CONFIGURATION
+
+#define TGS_DEFINE_AND_SET_GLOBAL(Name, Value) var/##Name = ##Value
+#define TGS_READ_GLOBAL(Name) global.##Name
+#define TGS_WRITE_GLOBAL(Name, Value) global.##Name = ##Value
+#define TGS_PROTECT_DATUM(Path)	//TODO
```
Comment (optimumtact) [inline code/__DEFINES/tgs.config.dm]: hahahah

**Comment by ninjanomnom** [comment]
Oh good I didn't want to have to do it

**Comment by MrStonedOne** [comment]
@Cyberboss travis

**Comment by Cyberboss** [comment]
PR isn't update fnr

**Pull request closed without merging**